Minor drape refactoring

pastk opened the pull request:
See commit messages.

biodranik reviewed (approved):
@vng PTAL

biodranik commented:
There are no new compile warnings, right?

pastk commented:
> There are no new compile warnings, right?

I didn't notice any.
BTW how to tell reliably which ones are new and which are old? :)

pastk commented:
@vng please review

vng commented on drape_frontend/stylist.cpp:
```diff
@@ -103,19 +103,19 @@ class Aggregator
   {
-    double depth = key.m_priority;
-    if (m_depthLayer != feature::LAYER_EMPTY)
+    float depth = key.m_priority;
```
Leave double here to keep precise calculations when multiply or some.

pastk replied:
Ok, but do we really need extra precision here?
	
All of the operands are whole numbers, so the remainder of the division will be a whole number too.
TBH all of this could be done with integers, maybe it'll be better to make m_depth `int` instead of float and cast to float only before passing it to e.g. `glsl::vec3`?
What do you think?

vng commented on drape_frontend/stylist.cpp:
```diff
@@ -143,7 +143,8 @@ class Aggregator
 
```
Make static_cast from double to float once here.

vng reviewed (commented):
There is one more entry of m_depth here:
if (rule.m_depth > maxPriority)

Should check and modify kMinPriority constant also.

pastk commented:
> There is one more entry of m_depth here: if (rule.m_depth > maxPriority)
> 
> Should check and modify kMinPriority constant also.

I've changed it too, thanks!

Also retained double for calculations for now, but see my comment above.

Rebased. PTAL!
